static/app/views/starfish/utils/sqlish/formatters/string.ts

```diff
@@ -3,25 +3,73 @@ import type {Token} from 'sentry/views/starfish/utils/sqlish/types';
 export function string(tokens: Token[]): string {
   let accumulator = '';
 
-  function contentize(content: Token): void {
-    if (Array.isArray(content.content)) {
-      content.content.forEach(contentize);
+  let precedingNonWhitespaceToken: Token | undefined = undefined;
+  let indentation: number = 0; // Track the current indent level
+  let parenthesisLevel: number = 0; // Tracks the current parenthesis nesting level
+  const indentationLevels: number[] = []; // Tracks the parenthesis nesting levels at which we've incremented the indentation
```

Comment:
took me a sec to figure out this is used like a map than as a list! Clever!

php arrays were ahead of their time.

Reply:
😂 only _sort of_, feels more like a stack to me, but you're onto something with the PHP comment